include/envoy/network/listen_socket.h

```diff
@@ -142,6 +151,9 @@ class ConnectionSocket : public virtual Socket {
    */
   virtual const Address::InstanceConstSharedPtr& remoteAddress() const PURE;
 
+  // Let the compiler know that we intend to use both.
+  using Socket::setLocalAddress;
```

Comment:
Hmm, should we make those two named slightly different? This confuses a bit.

Reply:
Fair enough. I considered it myself, but went with this for consistency. I'll change it.

Reply:
I'm not sure what to call it. I think it makes more sense to change the one in ConnectionSocket, since it's the one doing more than just setting the local address. I was thinking of calling it "restoreLocalAddress" and removing the `restored` parameter, but I was worried that we may want to transition from restored being true to restored being false, so I decided to leave the parameter, meaning that that name doesn't make sense.

For now I've gone with setConnectionLocalAddress. Any other ideas?

Reply:
Yeah changing one in ConnectionSocket is fine too, no strong preference either.